### PR TITLE
3.7 Documentation Update

### PR DESCRIPTION
**Pull request by BrentEaston**
Loads of Reference Manual updates
 - Reworked Properties page to add ALL currently defined properties
 - Reworked and simplified GamePiece page, added all current traits, Moved all descriptions to separate pages
 - Added/updated all trait pages
 - Ran all reference manual pages through optipng
 - Added a Concepts page indexed from front page.

### CODE REVIEW

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionSumCount.adoc [hunk 1]
```diff
@@ -522,7 +522,7 @@ Count the number of pieces on the Map named _East_ that have a non-blank value f
 ==== Attachment based functions
 Use these functions to *Sum* or *Count* values on pieces that are currently Attached to the source piece using the <<Attachment.adoc#top,Attachment>> trait.
 
-NOTE: The Attachment functionality allows pieces to be attached to themselves. If this is the case, then the current piece will be included in the Sum or Count returned by these functions.
+NOTE: The Attachment functionality allows pieces to be attached to themselves. If this is the case, then the current piece will be included in the Sum or Count returned by the these functions.
```
Comment (uckelman) [inline vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionSumCount.adoc:525]: `the` is a typo?

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/GamePiece.adoc [hunk 2]
```diff
@@ -196,22 +196,22 @@ Note, however, that Sub-Menu traits must always go *_below_* the traits that it
...
+|<<ActionButton.adoc#top,Action Button>> | Create a clickable button on a piece that issues a Key Command when clicked.
 
 |<<AreaOfEffect.adoc#top,Area Of Effect>> | Display a shaded area around a piece.
 
 |<<Attachment.adoc#top,Attachment>>| Create groups of 'Attached' counters that can easily interact with each other.
 
 | <<BasicName.adoc#top,Basic Name>> | Specify a primary name for a piece that over-rides the Basic Piece name.
 
-| <<BasicPiece.adoc#top,Basic Piece>> | Set the Base name and image for a piece.
+| <<#BasicPiece.adoc#top,Basic Piece>> | Set the Base name and image for a piece.
```
Comment (uckelman) [inline vassal-doc/src/main/readme-referencemanual/ReferenceManual/GamePiece.adoc:214]: Is the `#` added a typo?

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/Concepts.adoc [hunk 3]
```diff
@@ -79,74 +78,175 @@ A VASSAL Module is made up of two fundamentally different types of elements that
 
 [#components]
 === Components
-*Comp*
+VASSAL Components consist of the structural pieces that are used to implement a game. Some key Components used in nearly every module are
```
Comment (uckelman) [inline vassal-doc/src/main/readme-referencemanual/ReferenceManual/Concepts.adoc:81]: Missing `:`?

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionSumCount.adoc (hunk 1 above)
Comment (BrentEaston) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionSumCount.adoc:525]: Indeed

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/GamePiece.adoc (hunk 2 above)
Comment (BrentEaston) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/GamePiece.adoc:214]: Yep

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/Concepts.adoc (hunk 3 above)
Comment (BrentEaston) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/Concepts.adoc]: Fixed

**Comment by BrentEaston** [comment]
I will fix those conflicts

**Comment by BrentEaston** [comment]
Conflicts sorted.
I haven't finished the Doco update, but this is enough for beta 5, I've run out of steam!

**Review by uckelman — APPROVED** [review]

**Pull request merged**